presto-main/src/main/java/com/facebook/presto/execution/buffer/PagesSerde.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.execution.buffer;
+
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.block.BlockEncodingSerde;
+import io.airlift.compress.Compressor;
+import io.airlift.compress.Decompressor;
+import io.airlift.slice.DynamicSliceOutput;
+import io.airlift.slice.SliceOutput;
+import io.airlift.slice.Slices;
+
+import java.util.Optional;
+
+import static com.facebook.presto.execution.buffer.PageCompression.UNCOMPRESSED;
+import static com.facebook.presto.execution.buffer.PagesSerdeUtil.readRawPage;
+import static com.facebook.presto.execution.buffer.PagesSerdeUtil.writeRawPage;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.compress.lz4.Lz4RawCompressor.maxCompressedLength;
+import static java.lang.Math.toIntExact;
+import static java.util.Objects.requireNonNull;
+
+public class PagesSerde
+{
+    private static final double MINIMUM_COMPRESSION_RATIO = 0.8;
+
+    private final BlockEncodingSerde blockEncodingSerde;
+    private final Optional<Compressor> compressor;
+    private final Optional<Decompressor> decompressor;
+    private final PageCompression compression;
+
+    public PagesSerde(BlockEncodingSerde blockEncodingSerde, PageCompression compression, Optional<Compressor> compressor, Optional<Decompressor> decompressor)
```

Comment:
Why have `PageCompression compression` at all.  The value can be derived from the presence of the compressor and decompressor?